### ISSUE TITLE
[BUG] Image change is sometimes not detected

### ISSUE THREAD

**Issue by SendNukez** [JulienMaille/dribbblish-dynamic-theme#88]
Sometimes, when changing songs the colors aren't updated.

Maybe we should change the cover art listener to listen for attribute change on the first element of `.main-image-image.cover-art-image`, since the method we're using now is pretty tedious to implement it seems @JulienMaille?

**Comment by JulienMaille** [comment]
I'm in favor of a better event listener, but I have tried several trick which did not work in the past. Have you experimented ln this track ?

**Comment by SendNukez** [comment]
I was trying for a bit but didn't have time. I'm definitely going to try more. I was looking for a way to listen to attribute changes on the cover `<img>` so we can just listen to it and see when the src tag changes and go from there.

**Comment by JulienMaille** [comment]
The current hook works fine until Spotify decides to remove and recreate part of the DOM, including the playing now info and the cover elements. This happens when an advertisement is played but also randomly when switching playlists afaik

**Comment by SendNukez** [comment]
I created branch `[coverlistener](https://github.com/JulienMaille/dribbblish-dynamic-theme/tree/coverlistener)` for this. It seems to be working, but there was a part I removed where I don't really understand what it was for:
```js
waitForElement([".main-nowPlayingBar-left"], (queries) => {
    var observer = new MutationObserver(function (mutations) {
        mutations.forEach(function (mutation) {
            if (mutation.removedNodes.length > 0) coverListenerInstalled = false;
        });
    });
    observer.observe(queries[0], { childList: true });
});
```

**Comment by JulienMaille** [comment]
> I created branch [`coverlistener`](https://github.com/JulienMaille/dribbblish-dynamic-theme/tree/coverlistener) for this. It seems to be working, but there was a part I removed where I don't really understand what it was for:
> ```js
> waitForElement([".main-nowPlayingBar-left"], (queries) => {
>     var observer = new MutationObserver(function (mutations) {
>         mutations.forEach(function (mutation) {
>             if (mutation.removedNodes.length > 0) coverListenerInstalled = false;
>         });
>     });
>     observer.observe(queries[0], { childList: true });
> });
> ```

You can safety remove this. This was an attempt to handle edge cases where a big chunk of the DOM gets deleted when changing songs

**Comment by SendNukez** [comment]
So the new listener should be working. I haven't had any issues with it yet.

**Comment by SendNukez** [comment]
I'll push this to main, if there are any issues with it we can reopen / open a new issue.

**Comment by SendNukez** [comment]
Still happening with the new listener.

**Comment by JulienMaille** [comment]
What happens exactly? Does song changes skips a track or fully breaks?

**Comment by SendNukez** [comment]
Sometimes on startup the colors are just the default ones. The white on green. When skipping to the next song it changes. The initial reason for this issue was the same except changing the song wouldn't fix it, but now it does.

**Comment by JulienMaille** [comment]
Anything relevant in the logs?

**Comment by SendNukez** [comment]
Nope, nothing at all. I'm guessing that [this](https://github.com/JulienMaille/dribbblish-dynamic-theme/blob/c099d5c2057f4f721bc34e251afd1f9f36d01d84/src/js/main.js#L676) is not working how it's supposed to be. Maybe we have to not only check if the element exists, but also if the `src` attribute is present?

**Comment by JulienMaille** [comment]
Yes try to log the content of src here
https://github.com/JulienMaille/dribbblish-dynamic-theme/blob/c099d5c2057f4f721bc34e251afd1f9f36d01d84/src/js/main.js#L650

**Comment by SendNukez** [comment]
Actually, that might not be the problem. It sets the colors, but just not correctly.
![image](https://user-images.githubusercontent.com/29760988/139964574-30c1a98e-0850-4b8f-a3b1-942060d7ab0a.png)

**Comment by SendNukez** [comment]
This is very hard to debug, because it dosen't happen every time.

**Comment by SendNukez** [comment]
I think I just solved this and the solution is actually so simple I can't believe I didn't think of this. The problem was the image being detected but not loaded yet. I've opened Spotify like 50 times and it didn't break anymore. So this should (hopefully) be fixed in c3f1f23. If this isn't fixed still we'll just reopen again.
